src/mapping/query/path.rs

```diff
@@ -0,0 +1,199 @@
+use super::Function;
+use crate::{
+    event::{util::log::get_value, Event, PathIter, Value},
+    mapping::Result,
+};
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Debug)]
+pub(in crate::mapping) struct Path {
```

Comment:
It's kind of annoying how much logic gets duplicated with https://github.com/timberio/vector/blob/master/src/event/util/log/path_iter.rs but I understand why